std/algorithm/mutation.d

```diff
@@ -605,8 +606,10 @@ if ((isInputRange!Range && is(typeof(range.front = value)) ||
 @nogc @safe unittest
 {
     const(char)[] chars;
+    assert(chars.length == 0);
     static assert(!__traits(compiles, fill(chars, 'c')));
     wstring wchars;
+    assert(wchars.length == 0);
     static assert(!__traits(compiles, fill(wchars, wchar('c'))));
 }
```

Comment:
Executing the code will result in a compilation error since the code tests that fill does not compile in that particular case. You probably missed the '!'